bluer-tools/src/blumon.rs

```diff
@@ -76,13 +160,85 @@ impl DeviceMonitor {
                         } else {
                             self.show_device(data).await;
                         }
+
+                        let device = self.adapter.device(data.address)?;
+
+                        // Assuming `device.manufacturer_data().await?` returns a `HashMap<u16, Vec<u8>>`
+                        let manufacturer_data_map = device.manufacturer_data().await?.unwrap_or_default();
+
+                        // Convert the HashMap into a string representation
+                        let manufacturer_data_string = manufacturer_data_map.iter()
```

Comment:
Why convert it after all?
It is binaray data and can be stored as an integer array in JSON.

Reply:
I think the advantage of hexadecimal over decimal is that it improves the usability of the log file by making it easy to use command line tools to search for manufacture Company Identifier Code's which are given as hex values in the Bluetooth spec and also when using BlueZ tools. I modified the code to remove the ASCII character display which complicates parsing of the logs.

Reply:
True, but then JSON is not the right format. Its purpose is to be machine-readable with a minimal effort and doing custom unnecessary conversions is not helpful for that goal.

If you want a human-readable log, I would suggest adding another option that logs into a text file performing the conversions you described.